File: app/models/document.py
from sqlalchemy import Column, Integer, String, Text

from app.core.database import Base


class Document(Base):
    __tablename__ = "documents"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    authors = Column(String, nullable=False)
    year = Column(Integer, nullable=False)
    abstract = Column(Text, nullable=True)
    keywords = Column(String, nullable=True)
    category = Column(String, nullable=True)
    publisher = Column(String, nullable=True)
    isbn = Column(String, nullable=True)
    udk = Column(String, nullable=True)
    bbk = Column(String, nullable=True)
    rubrics = Column(Text, nullable=True)
    source_url = Column(String, nullable=True, unique=True, index=True)
    source_system = Column(String, nullable=True)
    external_id = Column(String, nullable=True)
    has_fulltext = Column(Integer, nullable=False, default=0)

File: frontend/src/types/document.ts
export interface Document {
  id: number
  title: string
  authors: string
  year: number
  abstract?: string
  keywords?: string
  category?: string
  publisher?: string
  isbn?: string
  udk?: string
  bbk?: string
  rubrics?: string
  source_url?: string
  source_system?: string
  external_id?: string
  has_fulltext?: number
}

export interface PopularDocument {
  id: number
  title: string
  authors: string
  year: number
  category: string
  interactions_count: number
}
